### ISSUE TITLE
EAP 2021.1 / (NameError) cannot load (ext) (org.jruby.ext.rbconfig.RbConfigLibrary)

### ISSUE THREAD

**Issue by ktomk** [asciidoctor/asciidoctor-intellij-plugin#664]
## Observed vs. expected behavior

Today I patched my PhpStorm EAP version and also updated the Asciidoctor plugin.

After restart the preview pane did not render the document any longer, the first line of the .adoc document is marked red.

The preview pane shows the following error message instead of rendering:

> unable to render AsciiDoc document
org.jruby.exceptions.NameError: (NameError) cannot load (ext) (org.jruby.ext.rbconfig.RbConfigLibrary)
>
> java.lang.IllegalArgumentException: Do not request resource from classloader using path with leading slash (path=/org/jruby/Ruby.class)
>
>(the full exception stack trace is available in the IDE's log file. Visit menu item 'Help | Show Log in Explorer' to see the log)

![Selection_128](https://user-images.githubusercontent.com/352517/107162891-5d8de500-69a6-11eb-93cf-0d8d73039aa3.png)

## Steps to reproduce

- Open .adoc file and its preview
- The .adoc file may even be empty
 

## Environment

Plugin Version: 0.32.16

IntelliJ Details: ____

PhpStorm 2021.1 EAP
Build #PS-211.5538.21, built on February 5, 2021
PhpStorm EAP User
Expiration date: March 7, 2021
Runtime version: 11.0.10+8-b1304.1 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-65-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1881M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: com.dubreuia (2.2.0), FrameSwitcher (3.25.1-211), com.paperetto.dash (3.3), org.intellij.plugins.hcl (0.7.10), name.kropp.intellij.makefile (211.5538.10), com.intellij.plugins.html.instantEditing (211.5538.10), com.kalessil.phpStorm.phpInspectionsUltimate (3.0.10), me.artspb.idea.eval.plugin (0.4), lv.midiana.misc.phpstorm-plugins.deep-keys (2021.01.05.001), ru.adelf.idea.dotenv (2021.1), org.asciidoctor.intellij.asciidoc (0.32.16)
Current Desktop: ubuntu:GNOME

**Comment by jvorhauer** [comment]
This happens in IntelliJ IDEA Ultimate 2021.1 EAP as well:

unable to render AsciiDoc document
org.jruby.exceptions.NameError: (NameError) cannot load (ext) (org.jruby.ext.rbconfig.RbConfigLibrary)

java.lang.IllegalArgumentException: Do not request resource from classloader using path with leading slash (path=/org/jruby/Ruby.class)

(the full exception stack trace is available in the IDE's log file. Visit menu item 'Help | Show Log in Explorer' to see the log)

**Comment by ahus1** [comment]
Thanks for reporting this; it seems the latest EAP breaks some Classloader things. Will look into this.

**Comment by ktomk** [comment]
To provide a bit more information to complete the report, here is my last known good configuration which I could migrate back to and can confirm working:

PhpStorm 2020.3.2
Build #PS-203.7148.74, built on January 27, 2021
Licensed to _redacted_
Subscription is active until March 18, 2021.
Runtime version: 11.0.9.1+11-b1145.77 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.4.0-65-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1881M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: com.dubreuia, FrameSwitcher, com.paperetto.dash, com.shellcheck, org.intellij.plugins.hcl, name.kropp.intellij.makefile, com.intellij.plugins.html.instantEditing, com.kalessil.phpStorm.phpInspectionsUltimate, me.artspb.idea.eval.plugin, lv.midiana.misc.phpstorm-plugins.deep-keys, ru.adelf.idea.dotenv, org.asciidoctor.intellij.asciidoc
Current Desktop: ubuntu:GNOME

It does not show the plugin versions, here is the version for AsciiDoc: 0.32.12

Updating to AsciiDoc: 0.32.16, also working.

**Comment by ahus1** [comment]
@ktomk - thanks. For now, I suggest downgrading to 2020.3.x until this issue is resolved.

**Comment by ktomk** [comment]
I have the versions in parallel, so if there is something to test ping me.

**Comment by ahus1** [comment]
I've published pre-release 0.32.17 that includes a fix. I applied an ugly workaround, and I asked JRuby upstream to fix it. Let's keep the fingers crossed there will be a releases. 

Please give it a try and let me know if this works for you.

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

**Comment by ktomk** [comment]
Just tested, can confirm fix, works as announced. +1 from my end. @ahus1

**Comment by ahus1** [comment]
Thanks for letting me know so quickly. I close this issue for now. I you discover other/new problems with the EAP, please open a new ticket.

**Comment by ktomk** [comment]
Thank you for this prompt fix. Can also confirm good for the 0.32.17 version against the last known good Phpstorm configuration. So counter-check from my end as well.
